test/org/opensolaris/opengrok/web/DirectoryListingTest.java

```diff
@@ -28,52 +28,69 @@
 import java.io.StringWriter;
 import java.text.SimpleDateFormat;
 import java.util.Arrays;
+import java.util.List;
 import javax.xml.parsers.DocumentBuilder;
 import javax.xml.parsers.DocumentBuilderFactory;
 import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
+import org.opensolaris.opengrok.history.RepositoryFactory;
 import org.opensolaris.opengrok.util.FileUtilities;
 import org.w3c.dom.Document;
 import org.w3c.dom.Element;
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
+
 import static org.junit.Assert.*;
-import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
-import org.opensolaris.opengrok.history.RepositoryFactory;
 
 /**
  * JUnit test to test that the DirectoryListing produce the expected result
  */
 public class DirectoryListingTest {
 
+    private static int DIRECTORY_INTERNAL_SIZE = -2;
+
     private File directory;
     private FileEntry[] entries;
     private SimpleDateFormat dateFormatter;
 
     class FileEntry implements Comparable {
+
         String name;
         String href;
         long lastModified;
-        int size; // If negative, don't check it.
+        int size;
+        List<FileEntry> subdirs;
 
         FileEntry() {
             dateFormatter = new SimpleDateFormat("dd-MMM-yyyy");
         }
 
-        FileEntry(String name, String href, long lastModified, int size) {
+        FileEntry(String name, String href, long lastModified, int size, List<FileEntry> subdirs) {
```

Comment:
pls keep the old method signature and make it a wrapper so that old calls do not have to be adjusted with the `null` argument.